src/controllers/users.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const usersServices = require('../services/users');
+
+const getAll = (_, res) => {
+  const users = usersServices.getAll();
+
+  res.send(users);
+};
+
+const getOne = (req, res) => {
+  const { id } = req.params;
+  const foundUser = usersServices.getById(id);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(foundUser);
+};
+
+const add = (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = usersServices.create(name);
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+const remove = (req, res) => {
+  const { id } = req.params;
+  const foundUser = usersServices.getById(id);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  usersServices.remove(id);
+  res.sendStatus(204);
+};
+
+const update = (req, res) => {
+  const { id } = req.params;
+  const foundUser = usersServices.getById(id);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const name = req.body.name;
+
+  if (typeof name !== 'string') {
+    res.sendStatus(422);
```

Comment:
Return 400 status code instead of 422. The request is invalid due to missing or incorrect parameters, not due to an unprocessable entity.
```suggestion
  if (typeof name !== 'string') {
    res.sendStatus(400);
```